src/engine/render.tsx

```diff
@@ -194,11 +212,12 @@ const RenderEngine = ({
   config,
   sitemap,
   components,
+  LinkComponent,
 }: IsomerBaseSchema): React.ReactNode => {
   if (components && components.length > 0) {
     const collatedComponents = components.map(
       (component: IsomerComponent, idx: number) => {
-        return <div key={idx}>{getComponent(component)}</div>
+        return <div key={idx}>{getComponent(component, LinkComponent)}</div>
```

Comment:
from what I gather, the design here is basically dependency injection where users of our components can decide what is the actual element used for links

just thinking out loud, perhaps in the future we could configure this once instead of passing it through a few functions, all the way to the component itself? for e.g I don't see why a layout would need to know about `LinkComponent`

Reply:
fair point, this PR demonstrates the use of NextLink injected but does not optimise the render logic as such. again, refactoring the render logic will be done in a separate PR